File: example/src/main/java/com/github/simonpercic/oklogexample/ui/MainComponent.java
package com.github.simonpercic.oklogexample.ui;

import com.github.simonpercic.oklogexample.AppComponent;
import com.github.simonpercic.oklogexample.ui.base.scope.PerActivity;

import dagger.Component;

/**
 * Main Activity's Dagger component.
 *
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
@PerActivity
@Component(
        dependencies = AppComponent.class,
        modules = MainModule.class
) interface MainComponent {

    void inject(MainActivity activity);
}

File: example/src/main/kotlin/com/github/simonpercic/oklogexample/ui/MainComponent.kt
package com.github.simonpercic.oklogexample.ui

import com.github.simonpercic.oklogexample.AppComponent
import com.github.simonpercic.oklogexample.ui.base.scope.PerActivity

import dagger.Component

/**
 * Main Activity's Dagger component.

 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(MainModule::class))
interface MainComponent {

    fun inject(activity: MainActivity)
}
